client/controllers/cavokeclientcontroller.cpp

```diff
@@ -120,27 +144,26 @@ void CavokeClientController::startQmlApplication(
         return;
     }
     qmlView->show();
-
-    // Test
-    emit gameModel->receiveUpdate("c++ -> qml working!");
 }
 
 void CavokeClientController::exitApplication() {
     testWindowView.close();
     joinGameView.close();
     gamesListView.close();
     createGameView.close();
+    settingsView.close();
     startView.close();
 }
-void CavokeClientController::startQmlByPath(const QString &path) {
-    currentQmlGameModel = new CavokeQmlGameModel(QUrl::fromUserInput(path));
+
+void CavokeClientController::startQmlByGameId(const QString &gameId) {
+    currentQmlGameModel = new CavokeQmlGameModel(
+        QUrl(cache_manager::get_cached_app_path(gameId)));
```

Comment:
без этого не работает под моей виндой. Но имхо лучше это сделать не здесь, а сразу `get_cached_app_path` и отдавать `QUrl`.

```suggestion
        QUrl::fromUserInput(cache_manager::get_cached_app_path(gameId)));
```